cmds/db-manager/main.go

```diff
@@ -166,44 +173,51 @@ func intAbs(x int) int {
 	return int(math.Abs(float64(x)))
 }
 
-func getCurrentDBVersion(crdbURI string) (string, error) {
+func createDatabaseIfNotExists(crdbURI string, database string) error {
 	crdb, err := dssCockroach.Dial(crdbURI)
 	defer func() {
 		crdb.Close()
 	}()
 	if err != nil {
-		return "", fmt.Errorf("Failed to dial CRDB while getting DB version: %v", err)
+		return fmt.Errorf("Failed to dial CRDB to check DB exists: %v", err)
 	}
-	// check if schema_versions table exists
-	const checkTableQuery = `
+	const checkDbQuery = `
 		SELECT EXISTS (
-  		SELECT *
-		  FROM information_schema.tables 
-		WHERE table_name = 'schema_versions'
-	)`
-	var (
-		version = "v0.0.0"
-		exists  bool
-	)
-
-	if err := crdb.QueryRow(checkTableQuery).Scan(&exists); err != nil {
-		return "", err
+			SELECT *
+				FROM pg_database 
+			WHERE datname = $1
+		)
+	`
+
+	var exists bool
+
+	if err := crdb.QueryRow(checkDbQuery, database).Scan(&exists); err != nil {
+		return err
 	}
 
 	if !exists {
-		return version, nil
-	}
-	// query for the schema version string
-	const getVersionQuery = `
-      SELECT schema_version 
-      FROM schema_versions
-	  WHERE onerow_enforcer = TRUE`
-	if err := crdb.QueryRow(getVersionQuery).Scan(&version); err != nil {
-		return "", err
+		log.Printf("Database \"%s\" doesn't exists, attempt to create", database)
+		createDB := fmt.Sprintf("CREATE DATABASE IF NOT EXISTS %s", database)
+		_, err := crdb.Exec(createDB)
+		if err != nil {
+			return fmt.Errorf("Failed to Create Database: %v", err)
+		}
 	}
-	// if for some reason the string returned is empty
-	if version == "" {
-		version = "v0.0.0"
+	return nil
+}
+
+func getCurrentDBVersion(crdbURI string, database string) (string, error) {
+	crdb, err := dssCockroach.Dial(crdbURI)
+	defer func() {
+		crdb.Close()
+	}()
+	if err != nil {
+		return "", fmt.Errorf("Failed to dial CRDB while getting DB version: %v", err)
+	}
+	version, err := dssCockroach.GetVersion(context.Background(), crdb, database)
+	if err != nil && version == "" {
```

Comment:
If `err != nil`, it shouldn't matter what version is; we can remove `&& version == ""`.

Reply:
so I had to modify the GetVersion method itself, so that if the schema_versions table doesn't exists it should return v0.0.0 by default unless the it is a different error

Reply:
@vosst could you take a look at this?  IIUC, idiomatic Go returns either err or value, but never both at the same time.